test/runnable/e7804.d

```diff
@@ -70,7 +70,7 @@ TmpPrm!(__traits(getMember, Foo, "MyInt")) tpt = TmpPrm!(__traits(getMember, Foo
         assert(vm[0](42) == 42);
         alias attribs = __traits(getAttributes, Class);
         assert(attribs.length == 2);
-        assert(attribs[0] is Foo);
```

Comment:
:scream: I mentioned that the other day : when something works, people rarely try to prove that it should not so it's actually a chance that GDC works differently because this had discretely "slipped in" during the review process.

Reply:
It wouldn't be the first time.  A few years back, dmd quite happily compiled:
```
bool boolFunc();
void voidFunc();
if (boolFunc() || voidFunc())
{
  // ...
}
```
No one noticed until gdc tried to compile some newly introduced code.